core/src/main/clojure/xtdb/expression.clj

```diff
@@ -1630,9 +1639,11 @@
                                           ~@writer-bindings
                                           row-count# (.getRowCount ~rel-sym)]
                                       (dotimes [~idx-sym row-count#]
-                                        ~(continue (fn [t c]
-                                                     (write-value-out! t c))))))
-
+                                        (try
+                                          ~(continue (fn [t c]
+                                                       (write-value-out! t c)))
+                                          (catch ArithmeticException e#
```

Comment:
probably needn't include this in the code generation - just outside should suffice?